extensions/dim-rename-kafka8/src/main/java/io/druid/server/namespace/KafkaExtractionManager.java

```diff
@@ -0,0 +1,239 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.namespace;
+
+import com.google.common.base.Function;
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.metamx.common.IAE;
+import com.metamx.common.ISE;
+import com.metamx.common.StringUtils;
+import com.metamx.common.lifecycle.LifecycleStart;
+import com.metamx.common.lifecycle.LifecycleStop;
+import com.metamx.common.logger.Logger;
+import io.druid.guice.ManageLifecycle;
+import io.druid.query.extraction.namespace.KafkaExtractionNamespace;
+import io.druid.server.namespace.cache.NamespaceExtractionCacheManager;
+import kafka.consumer.ConsumerConfig;
+import kafka.consumer.ConsumerIterator;
+import kafka.consumer.KafkaStream;
+import kafka.consumer.Whitelist;
+import kafka.javaapi.consumer.ConsumerConnector;
+import kafka.message.MessageAndMetadata;
+import kafka.serializer.Decoder;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Pattern;
+
+/**
+ *
+ */
+@ManageLifecycle
+public class KafkaExtractionManager
+{
+  private static final Logger log = new Logger(KafkaExtractionManager.class);
+
+  private final Properties kafkaProperties;
+  private final ConcurrentMap<String, Function<String, String>> fnCache;
+  private final Collection<ListenableFuture<?>> futures = new ConcurrentLinkedQueue<>();
+  private final ListeningExecutorService executorService = MoreExecutors.listeningDecorator(
+      Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder()
+              .setNameFormat("kafka-rename-consumer-%d")
+              .setDaemon(true)
+              .setPriority(Thread.MIN_PRIORITY)
+              .build()
+      )
+  );
+  private final NamespaceExtractionCacheManager cacheManager;
+  private final AtomicInteger startedPipes = new AtomicInteger(0);
+  private final AtomicLong events = new AtomicLong(0l);
+
+  // Bindings in KafkaExtractionNamespaceModule
+  @Inject
+  public KafkaExtractionManager(
+      @Named("renameKafkaProperties") Properties kafkaProperties,
+      NamespaceExtractionCacheManager cacheManager,
+      @Named("io.druid.server.namespace.NamespacedExtractionModule")
```

Comment:
is "io.druid.server.namespace.NamespacedExtractionModule" supposed to match with name of class NamespacedExtractionModule for some reason and needs to be changed if we (for some hypothetical reason) rename that class?
otherwise, wouldn't a name like "namespaceMappingsCache" be more meaningful?